### ISSUE TITLE
Another geometry optimization crash

### ISSUE THREAD

**Issue by susilehtola** [psi4/psi4#2213]
Here's another example of a hard geometry optimization problem.

Psi4 1.3.2 crashes/segfaults; @psi-rking's pyoptking also gives up.
```
import optking
memory 8 gb

molecule {
Mn       -1.19858918223104   -0.08207133498850   -0.13371354096607
N        -0.41561138019044    1.27599387849124    0.21223826269367
O         0.14600552164730    2.25554129585451    0.46379448621401
C        -2.77325495017405    0.77903635981534   -0.46993705135304
C        -0.09036086226268   -0.50949483218465   -1.52013938052672
O         0.61690491558852   -0.75191770614607   -2.38035917189752
O        -3.74228303591446    1.34350620162321   -0.67446015379883
C        -2.08944770913093   -1.62303880517967   -0.52670173551821
C        -0.89694974294004   -0.80918135325017    1.51325120318508
O        -2.64631747328247   -2.58917480422090   -0.77666675977250
O        -0.69173610111097   -1.23796889981931    2.54918384173987
}

set basis STO-3G
set geom_maxiter 250

# Use a (75,302) quadrature grid
set dft_radial_points 75
set dft_spherical_points 302

# Psi4 has problems converging the first SCF with PBE so we initialize
# with Hartree-Fock orbitals
set df_basis_scf def2-svp-jkfit
energy('scf')
set df_basis_scf def2-svp-jfit
set guess read

optimize('pbe') # Crashes in psi4 1.3.2
#optking.optimize_psi4('pbe') # Also crashes
```

**Comment by AlexHeide** [comment]
I'm taking a look at this in pyoptking and hit the scf issue described above. @loriab `"guess" : "read"` is being sent to psi4 in the `AtomicInput`. Is `read` expected to work with a standard energy call followed by a call through `run_qcschema`? 

I get:
`Unable to find file 180, defaulting to SAD guess.` 

If psi4 is looking in `$PSI_SCRATCH` I don't see file 180 there.

**Comment by loriab** [comment]
No, guess=read is specifically disabled https://github.com/psi4/psi4/blob/master/psi4/driver/p4util/python_helpers.py#L451 .

As to how to run the calc, there's ways to deposit and collect scratch files in a qcng.compute run, and that could be done with the npy file and the optimize() fn or the optking qcng procedure. Ultimately, the optking qcng procedure should set the orbitals protocol so that it collects orbitals from gradients and can reseed them for the next gradient. The barrier to that right now is that psi4 can write the QCSchema Wavefunction, but it can't reinitialize a psi4.Wfn from it. (It's a BasisSet coeff issue I noticed last summer but haven't been able to follow up on.)

So this input is something we don't want to prevent, but it would need some hacking at the present, and it might hit some other problem along the way.

**Comment by susilehtola** [comment]
The real problem here is that both the old and the new optking crash because they are unable to find out a coordinate system for the molecule.

**Comment by AlexHeide** [comment]
Yep I agree. Thank you for linking this issue to the optking repo as well and bringing to our attention. I'm trying to look at more github issues for optking since you all have put so much effort into nicely labeling everything for us - still looking at this one. Any more discussion on the actual torsion issue will likely be at psi-rking/optking#63  or in slack once I've actually done some poking around. I'll make sure to reference the eventual fix here.

The quick fix for running the optimization is to use Cartesian coordinates.

For debugging purposes I'll just be running HF, good to know that guess=read is not currently supported - thank you.

**Comment by psi-rking** [comment]
Today I looked at this ~D3h Mn(NO)(CO)4 complex.  It is pathologically interesting.
1. There are no reasonable torsions defined among the bonded atoms in this molecule, as each of the 5 spokes out from the metal involve 2 ~collinear atoms.
2. The reason that optking was trying to use torsions is that the input geometry is so crowded that the covalent radii * 1.3 formula was bonding the Mn to the atom once removed (beta?).  That is, in the linear Mn-N-O, both Mn-N and Mn-O bonds were being created.  So I tried covalent_connect=1.2, which results in only linear bends being defined and no torsions.  This raises the question of whether we should use something beyond a linear scaling, or count on the user to provide a reasonable structure.  (The clean tool in Spartan lengthened the Mn-N bond by 25%.). For the moment, I am not changing. 
3. Using these coordinates (all stretches and bends), the first step blew up to 100 Angstroms Cartesian step.  This was not due to forces, which were large but not that ridiculous.
4. The numerical test of the B matrix passed. (a non-trivial accomplishment for this thing)
5. I figured out that the pseudoinverse of (B B^T) was blowing up, due to inversion of very small values.  It is customary to invert only those eigenvalues whose magnitude is >1e-10 or so.  A numerical problem was introduced when my custom linear algebra function was replaced by numpy.linalg.pinv which uses a default of 1e-14.  I'm actually surprised if this threshold isn't causing problems more commonly.  Anyway, I reduced this to 1e-12 and added a keyword to reduce it further.  More testing is needed to see if we can make the default larger.  This particular case needs 1e-8 and then the optimization iterates successfully downhill.  The geometry is very far from equilibrium and may be changing electronic states, so I'm not pursuing its STO-3G to a minimum whatever that might be.
So I am going to commit some related things but I'm not going to attend to this particular case further.
